### ISSUE TITLE
craft-application 4.2.6 depends on requests ~=2.32.0

### ISSUE THREAD

**Issue by vpa1977** [canonical/craft-application#511]
### Bug Description

craft-application 4.2.6 depends on requests 2.32, which break requests-unixsocket  https://github.com/msabramo/requests-unixsocket/pull/72, breaking rockcraft.

### To Reproduce

requirements.txt
```
craft-application==4.2.6
craft-parts==2.1.2
craft-providers==2.0.4
requests==2.31.0
```

Fails to install with the following error:
```
ERROR: Cannot install craft-application==4.2.6 and requests==2.31.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested requests==2.31.0
    craft-application 4.2.6 depends on requests~=2.32
```

### Relevant log output

```shell
2024-10-08 16:14:17.838 During handling of the above exception, another exception occurred:
2024-10-08 16:14:17.838 Traceback (most recent call last):
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/craft_application/application.py", line 568, in run
2024-10-08 16:14:17.838     return_code = self._run_inner()
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/craft_application/application.py", line 552, in _run_inner
2024-10-08 16:14:17.838     self.run_managed(platform, build_for)
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/craft_application/application.py", line 370, in run_managed
2024-10-08 16:14:17.838     with self.services.provider.instance(
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/usr/lib/python3.10/contextlib.py", line 135, in __enter__
2024-10-08 16:14:17.838     return next(self.gen)
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/craft_application/services/provider.py", line 134, in instance
2024-10-08 16:14:17.838     provider.ensure_provider_is_available()
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/craft_providers/lxd/lxd_provider.py", line 78, in ensure_provider_is_available
2024-10-08 16:14:17.838     if not is_installed():
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/craft_providers/lxd/installer.py", line 122, in is_installed
2024-10-08 16:14:17.838     snap_info = requests_unixsocket.get(url=url, params={"select": "enabled"})
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/requests_unixsocket/__init__.py", line 51, in get
2024-10-08 16:14:17.838     return request('get', url, **kwargs)
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/requests_unixsocket/__init__.py", line 46, in request
2024-10-08 16:14:17.838     return session.request(method=method, url=url, **kwargs)
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
2024-10-08 16:14:17.838     resp = self.send(prep, **send_kwargs)
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
2024-10-08 16:14:17.838     r = adapter.send(request, **kwargs)
2024-10-08 16:14:17.838   File "/snap/rockcraft/x1/lib/python3.10/site-packages/requests/adapters.py", line 534, in send
2024-10-08 16:14:17.838     raise InvalidURL(e, request=request)
2024-10-08 16:14:17.838 requests.exceptions.InvalidURL: Not supported URL scheme http+unix
``
```

**Comment by mr-cal** [comment]
> Hi! This unpinning of the requests version was intentional, though we probably shouldn't have set it to ~=2.32 in the hotfix. @mr-cal should we release a 4.2.7 than unpins it (allowing apps to use 2.31 still)?

I see, I agree - I didn't need to pin it like that in a hotfix.

To be clear, do you mean unpin it in the hotfix but leave it as-is in main?

**Comment by lengau** [comment]
@mr-cal I think it's fair for us to unpin in general and just leave that up to apps.

**Comment by mr-cal** [comment]
Fixed in craft-application 4.2.7.
